libraries/ESP8266WiFi/src/include/UdpContext.h

```diff
@@ -390,7 +391,16 @@ class UdpContext
         return size;
     }
 
-    bool send(const ip_addr_t* addr = 0, uint16_t port = 0)
+    void cancelBuffer ()
+    {
+        if (_tx_buf_head)
+            pbuf_free(_tx_buf_head);
+        _tx_buf_head = 0;
+        _tx_buf_cur = 0;
+        _tx_buf_offset = 0;
+    }
+
+    err_t trySend(const ip_addr_t* addr = 0, uint16_t port = 0, bool keepBuffer = true)
```

Comment:
`trySend()` in conjunction with `sendTimeout()` handle error cases, including OOM cases.
`trySend()` has been moved to the private section leaving only `sendTimeout()` in the user-facing API.

> How are the pbufs that were already received cleaned up?

lwIP handles its part and returns an error case 
```
// trySend() is now a private function member
        err_t err = udp_sendto(_pcb, tx_copy, addr, port);
        ...
        pbuf_free(tx_copy);
        if (err == ERR_OK)
            cancelBuffer(); // no error: get rid of buffer
        return err;  <- sendTimeout() will retry or cancel buffer after timeout in case of error
```